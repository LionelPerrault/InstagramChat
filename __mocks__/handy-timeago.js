export default () => 'a month ago'
