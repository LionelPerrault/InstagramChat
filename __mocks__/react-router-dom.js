export const Link = 'Link'
export const Switch = 'Switch'
export const Redirect = 'Redirect'
export const Route = 'Route'
export const NavLink = 'NavLink'
