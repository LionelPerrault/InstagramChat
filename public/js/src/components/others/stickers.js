import React from 'react'
import { FadeIn } from 'animate-components'
import { Scrollbars } from 'react-custom-scrollbars'
import Spinner from './spinner'
import { post } from 'axios'
import $ from 'jquery'

export default class Stickers extends React.Component {

  state = {
    stickers: [],
    loading: true,
    selectedSticker: ''
  }

  componentDidMount = async () => {
    let { data: stickers } = await post('/api/get-stickers')
    this.setState({ stickers, loading: false })
  }

  back = e => {
    e.preventDefault()
    this.props.back()
  }

  selectSticker = e => {
    let sticker = $(`[data-sticker="sticker-${e}"]`)
    sticker.siblings().removeClass('sti_img_active')
    sticker.addClass('sti_img_active')
    $('.btn_select_sticker').focus()
    this.setState({ selectedSticker: e })
  }

  chooseSticker = async e => {
    e.preventDefault()
    $('.btn_select_sticker').addClass('a_disabled')

    let
      { selectedSticker } = this.state,
      { type, back, stickerComment, stickerMessage } = this.props

    if (type == 'comment') {
      stickerComment(selectedSticker)
    } else {
      stickerMessage(selectedSticker)
    }

    back()
  }

  render() {
    let
      { loading, stickers } = this.state,
      map_stickers = stickers.map(s =>
        <img
          key={s}
          src={`/images/stickers/${s}`} className='sti_img'
          data-sticker={`sticker-${s}`}
          onClick={() => this.selectSticker(s)}
        />
      )

    return (
      <div class='stickers_modal modal modal_big' >

        <FadeIn duration='300ms' >
          <div className='modal_header'>
            <span className='title' >Choose a Sticker</span>
          </div>

          <Scrollbars style={{ height: 450 }} className='modal_middle' >

            { loading ? <Spinner/> : null }

            <div className={`modal_main ${loading ? 'cLoading' : ''}`}>
              { map_stickers }
            </div>

          </Scrollbars>

          <div className='modal_bottom'>
            <a href='#' className='sec_btn' onClick={this.back} >Back</a>
            <a
              href='#'
              className={`pri_btn btn_select_sticker ${loading ? 'a_disabled' : ''}`}
              onClick={this.chooseSticker}
            >Choose</a>
          </div>
        </FadeIn>

      </div>
    )
  }
}
