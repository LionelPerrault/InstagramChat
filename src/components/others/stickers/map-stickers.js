import React from 'react'
import PropTypes from 'prop-types'
import { FadeIn } from 'animate-components'

const MapStickers = ({ stickers, selectSticker }) => {
  let map_stickers = stickers.map(s => (
    <img
      key={s}
      src={`/images/stickers/${s}`}
      className="sti_img"
      data-sticker={`sticker-${s}`}
      onClick={() => selectSticker(s)}
    />
  ))

  return <FadeIn duration="300ms">{map_stickers}</FadeIn>
}

MapStickers.propTypes = {
  stickers: PropTypes.arrayOf(PropTypes.string).isRequired,
  selectSticker: PropTypes.func.isRequired,
}

export default MapStickers
