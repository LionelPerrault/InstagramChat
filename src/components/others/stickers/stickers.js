import React, { Component, Fragment } from 'react'
import { FadeIn } from 'animate-components'
import { Scrollbars } from 'react-custom-scrollbars'
import { post } from 'axios'
import PropTypes from 'prop-types'
import MapStickers from './map-stickers'
import d from '../../../utils/API/DOM'
import ModalHeader from '../modal/modal-header'
import ModalBack from '../modal/modal-back'
import IsLoading from '../isLoading'
import { cLoading } from '../../../utils/utils'
import PrimaryButton from '../button/primary-btn'
import Overlay from '../overlay'
import classNames from 'classnames'

export default class Stickers extends Component {
  state = {
    stickers: [],
    loading: true,
    selectedSticker: '',
  }

  componentDidMount = async () => {
    let { data: stickers } = await post('/api/get-stickers')
    this.setState({
      stickers,
      loading: false,
    })
  }

  selectSticker = e => {
    let sticker = new d(`[data-sticker="sticker-${e}"]`)
    let siblings = new d('.sti_img')

    siblings.removeClass('sti_img_active')
    sticker.addClass('sti_img_active')
    new d('.btn_select_sticker').focus()

    this.setState({ selectedSticker: e })
  }

  chooseSticker = e => {
    e.preventDefault()
    new d('.btn_select_sticker').addClass('a_disabled')

    let { selectedSticker } = this.state
    let { back, stickerSelected } = this.props
    stickerSelected(selectedSticker)
    back()
  }

  render() {
    let { loading, stickers } = this.state
    let { back } = this.props

    return (
      <Fragment>
        <Overlay />

        <div className="stickers_modal modal modal_big">
          <FadeIn duration="300ms">
            <ModalHeader title="Choose a sticker" />

            <Scrollbars style={{ height: 450 }} className="modal_middle">
              <IsLoading loading={loading} />

              <div className={classNames('modal_main', cLoading(loading))}>
                <MapStickers
                  stickers={stickers}
                  selectSticker={this.selectSticker}
                />
              </div>
            </Scrollbars>

            <div className="modal_bottom">
              <ModalBack back={back} btnType="secondary" />
              <PrimaryButton
                label="Choose"
                onClick={this.chooseSticker}
                extraClass="btn_select_sticker"
                disabled={loading}
              />
            </div>
          </FadeIn>
        </div>
      </Fragment>
    )
  }
}

Stickers.propTypes = {
  back: PropTypes.func.isRequired,
  stickerSelected: PropTypes.func,
}
