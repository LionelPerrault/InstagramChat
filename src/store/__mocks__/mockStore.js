import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
