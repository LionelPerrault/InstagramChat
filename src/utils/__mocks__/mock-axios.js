import AxiosAdaper from 'axios-mock-adapter'
import axios from 'axios'
const mockAxios = new AxiosAdaper(axios)

/**
 * Mocks Axios requests.
 *
 * @param {String} endpoint URL to request data from.
 * @param {any} data Fake data to return.
 */
export const mockAxiosRequest = (endpoint, data) =>
  mockAxios.onPost(`/api/${endpoint}`).reply(200, data)
