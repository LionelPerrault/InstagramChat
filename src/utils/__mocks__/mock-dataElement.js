/**
 * Creates a fake div.data with required data attributes for use in tests.
 * @returns {HTMLDivElement} dataElementDIV
 */
const MockDataElement = () => {
  let dataElement = document.createElement('div')
  dataElement.setAttribute('class', 'data')
  dataElement.setAttribute('data-session', '24')
  dataElement.setAttribute('data-username', 'takkar')
  dataElement.setAttribute('data-email-verified', 'no')
  dataElement.setAttribute('data-isadmin', 'false')
  document.body.prepend(dataElement)

  return dataElement
}

export default MockDataElement
