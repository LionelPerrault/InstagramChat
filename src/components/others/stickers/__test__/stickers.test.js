import React from 'react'
import Stickers from '../stickers'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'
import stickers from './stickers-mockArray'
import { mockAxiosRequest } from '../../../../utils/__mocks__/mock-axios'
import stickersMockArray from './stickers-mockArray'

describe('Stickers Component', () => {
  const mockFn = jest.fn()
  const comp = <Stickers back={mockFn} stickerSelected={mockFn} />

  mockAxiosRequest('get-stickers', stickersMockArray)

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should hide spinner when loading=false', () => {
    const wrapper = shallow(comp)
    wrapper.setState({ loading: false })

    expect(wrapper.find('IsLoading').prop('loading')).toBe(false)
  })

  it('should get stickers from API in componentDidMount', async () => {
    const wrapper = shallow(comp)
    await wrapper.instance().componentDidMount()
    expect(wrapper.state().stickers).toBeArray()
    expect(wrapper.state().stickers).toIncludeAnyMembers([stickers[0]])
  })

  it('should perform some click actions', async () => {
    const wrapper = shallow(comp)
    await wrapper.instance().componentDidMount()

    wrapper.setState({
      loading: false,
      // we assume first sticker is selected and it's file path is return by stickerSelected prop
      selectedSticker: wrapper.state().stickers[0],
    })

    wrapper.find('ModalBack').simulate('click')
    wrapper.find('PrimaryButton').simulate('click', { preventDefault() {} })
  })
})
