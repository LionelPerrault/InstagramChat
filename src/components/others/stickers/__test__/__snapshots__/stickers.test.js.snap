// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stickers Component should match snapshot 1`] = `
Array [
  <div
    className="overlay"
    onClick={null}
    style={
      Object {
        "cursor": "inherit",
        "opacity": 0.5,
      }
    }
  />,
  <div
    className="stickers_modal modal modal_big"
  >
    <div
      style={
        Object {
          "animation": "bcCCNc 300ms ease 0s 1 normal none running",
          "backfaceVisibility": "visible",
        }
      }
    >
      <div
        className="modal_header"
      >
        <span
          className="title"
        >
          Choose a sticker
        </span>
      </div>
      <div
        className="modal_middle"
        style={
          Object {
            "height": 450,
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "bottom": 0,
              "left": 0,
              "marginBottom": 0,
              "marginRight": 0,
              "overflow": "scroll",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <div
            className="spinner"
          >
            <span />
            <span />
            <span />
          </div>
          <div
            className="modal_main cLoading"
          >
            <div
              style={
                Object {
                  "animation": "bcCCNc 300ms ease 0s 1 normal none running",
                  "backfaceVisibility": "visible",
                }
              }
            />
          </div>
        </div>
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "height": 6,
              "left": 2,
              "position": "absolute",
              "right": 2,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": "inherit",
                "cursor": "pointer",
                "display": "block",
                "height": "100%",
                "position": "relative",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "position": "absolute",
              "right": 2,
              "top": 2,
              "width": 6,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": "inherit",
                "cursor": "pointer",
                "display": "block",
                "position": "relative",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
      <div
        className="modal_bottom"
      >
        <a
          className="sec_btn"
          href="#"
          onClick={[Function]}
        >
          Back
        </a>
        <a
          className="pri_btn btn_select_sticker a_disabled"
          href="#"
          onClick={[Function]}
        >
          Choose
        </a>
      </div>
    </div>
  </div>,
]
`;
